### PR TITLE
Add missing settings hmac_jwkset and hmac_key for JWT realm to security-settings.asciidoc

### PR DESCRIPTION
**Pull request by justincr-elastic**
Elasticsearch docs were updated to add JWT Realm in PR #85189. To prevent holding up merging of that PR, this small follow-up PR was created immediately afterwards.

These two settings are in the JWT Realm page, but missing from the Security Settings reference page.
- hmac_jwkset
- hmac_key

This Docs Preview link is updated after each commit, and remains available for a short time:
- https://elasticsearch_86085.docs-preview.app.elstc.co/diff

### CODE REVIEW

File: docs/reference/settings/security-settings.asciidoc [hunk 1]
```diff
@@ -2174,7 +2174,29 @@ rotated the signing keys.
...
+the JWT Realm uses for verifying token signatures.
+This format is compatible with multiple keys and optional attributes.
+Only one of `hmac_jwkset` or `hmac_key` can be used simultaneously.
+Using the HMAC JWK Set is preferred.
+// end::jwt-hmac-jwkset-tag[]
+
+// tag::jwt-hmac-key-tag[]
+`hmac_key` {ess-icon}::
+(<<secure-settings,Secure>>)
+The contents of a JSON Web Key with the secret key material that
```
Comment (lockewritesdocs) [inline docs/reference/settings/security-settings.asciidoc:2192]: The `hmac_jwkset` contains "The file contents of  JWK Set", whereas the `hmac_key` indicates only "The contents". Is the `hmac_key` also a file? I'm trying to better understand the distinction between these settings other than what they support.

Comment (justincr-elastic) [reply docs/reference/settings/security-settings.asciidoc:2192]: Both can be `The contents`, so remove `file`. I was implying they would be loaded via the different elasticsearch-keystore commands listed in the `[JWT authentication](https://elasticsearch_85189.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/jwt-realm.html)` page.

HMAC JWKSet loaded from a file
`bin/elasticsearch-keystore add-file xpack.security.authc.realms.jwt.jwt1.hmac_jwkset <path>`

HMAC Key loaded from a string on the command line
`bin/elasticsearch-keystore add xpack.security.authc.realms.jwt.jwt1.client_authentication.shared_secret`

File: docs/reference/settings/security-settings.asciidoc [hunk 2]
```diff
@@ -2174,7 +2174,29 @@ rotated the signing keys.
...
+// end::jwt-pkc-jwkset-path-tag[]
+
+// tag::jwt-hmac-jwkset-tag[]
+`hmac_jwkset` {ess-icon}::
+(<<secure-settings,Secure>>)
+The contents of a JSON Web Key Set with the secret key material that
+the JWT Realm uses for verifying token signatures.
+This format is compatible with multiple keys and optional attributes.
+Only one of `hmac_jwkset` or `hmac_key` can be used simultaneously.
+Using the HMAC JWK Set is preferred.
```
Comment (lockewritesdocs) [inline docs/reference/settings/security-settings.asciidoc:2186]:
```suggestion
Contents of a JSON Web Key Set (JWKS), including the secret key that the JWT 
realm uses to verify token signatures. This format supports multiple keys and 
optional attributes, and is preferred over the `hmac_key` setting. Cannot be
used in conjunction with the `hmac_key` setting.

```

**Review by lockewritesdocs — COMMENTED** [review]
Left some suggestions for updating the settings descriptions.

File: docs/reference/settings/security-settings.asciidoc [hunk 3]
```diff
@@ -2174,7 +2174,29 @@ rotated the signing keys.
...
+// tag::jwt-hmac-key-tag[]
+`hmac_key` {ess-icon}::
+(<<secure-settings,Secure>>)
+The contents of a JSON Web Key with the secret key material that
+the JWT Realm uses for verifying token signatures.
+This format is compatible with OIDC HMAC key encoding, where
+the HMAC key bytes are the UTF-8 encoding of the UNICODE characters.
+This format only supports a single key with no attributes.
+Only one of `hmac_jwkset` or `hmac_key` can be used simultaneously.
+Using the HMAC JWK Set is preferred.
```
[the remark this answers is not in the document]
Comment (justincr-elastic) [reply docs/reference/settings/security-settings.asciidoc]: @lockewritesdocs the rewording of the OIDC HMAC key format made the description not totally accurate. I accepted the suggestion, and I tried to rephrase it. Have a look to see if the update is OK.

**Review by lockewritesdocs — APPROVED** [review]
LGTM -- thanks for iterating @justincr-elastic!

**Pull request merged**